### PR TITLE
RSDK-9441 clean up app wrappers

### PR DESCRIPTION
**Pull request by purplenicole730**
This PR mainly focuses on basic cleanup for app wrappers. Naming and return types were standardized to follow the rest of the app wrappers. Note that these changes do not necessarily mean that something was done wrong previously.

Changes:
- return pointers where pointers are returned from proto
- name response shadow types `<Method>Response`
- convert `anypb.Any` types to Go standard types
- call `MQL` function with queries
- remove pointers from types that are already references
- return errors where applicable
- check for errors in tests
- return proto type for Google Status object

### CODE REVIEW

File: app/mltraining_client_test.go [hunk 1]
```diff
@@ -39,12 +39,7 @@ var (
 		ErrorStatus: &ErrorStatus{
 			Code:    code,
 			Message: message,
-			Details: []*GenericProtoMessage{
-				{
-					TypeURL: siteURL,
-					Value:   byteData,
-				},
-			},
+			Details: []interface{}(nil),
```
Comment (stuqdog) [inline app/mltraining_client_test.go:42]: (q) curious why we move to an empty `Details` here? I think having a specific type in there to reflect what would typically go here would be helpful.

File: app/mltraining_client.go [hunk 2]
```diff
@@ -34,7 +35,7 @@ const (
 type ErrorStatus struct {
 	Code    int
 	Message string
-	Details []*GenericProtoMessage
+	Details []interface{}
```
Comment (stuqdog) [inline app/mltraining_client.go:38]: (q) I think I'm slightly confused about this. Can it really be _anything_ that goes into the details?

**Review by stuqdog — COMMENTED** [review]
Nice, this looks good! I think I've got some confusion around the changes to the `GenericProtoMessage` type but otherwise this looks good to me.

File: app/mltraining_client.go (hunk 2 above)
Comment (purplenicole730) [reply app/mltraining_client.go:38]: There doesn't seem to be much clarification on what `Details` really is. The other SDKs leave it at the `Any` proto type. Converting `Any` types to `interface{}` is also something I see elsewhere in the RDK.

The only information I really see is:
```
	// A list of messages that carry the error details.  There is a common set of
	// message types for APIs to use.
```

**Review by stuqdog — APPROVED** [review]

**Pull request merged**